virtme/commands/run.py

```diff
@@ -968,6 +969,46 @@ def cleanup_console_script():
     qemuargs.extend(["-device", "vhost-vsock-pci,guest-cid=%d" % args.port])
 
 
+def ssh_client(args):
+    user = f"{args.user}@" if args.user else ""
+
+    if args.remote_cmd is not None:
+        exec_escaped = args.remote_cmd.replace('"', '\\"')
+        remote_cmd = f'bash -c "{exec_escaped}"'
+    else:
+        remote_cmd = ""
+
+    cmd = ["ssh", "-p", str(args.port), f"{user}localhost"] + (
+        [remote_cmd] if remote_cmd else []
+    )
+    if args.dry_run:
+        print(" ".join(cmd))
+    else:
+        os.execvp("ssh", cmd)
+
+
+def ssh_server(args, arch, qemuargs, kernelargs):
+    # Check if we need to generate the ssh host keys for the guest.
+    ssh_key_dir = f"{CACHE_DIR}/.ssh"
```

Comment:
you might want to prefix it with `args.root` if defined, no?

Reply:
In the future definitely yes, but for now `--root` can't be used with `--ssh` (for the potential permissions limitations). This should also be solved once we introduce the generic shared cache dir between host and guest.